consensus/src/dag/dag_driver.rs

```diff
@@ -77,16 +83,22 @@ impl DagDriver {
             storage,
             order_rule,
             fetch_requester,
-        }
-    }
+        };
 
-    pub fn try_enter_new_round(&mut self) {
-        // In case of a new epoch, kickstart building the DAG by entering the next round
-        // without any parents.
-        if self.current_round == 0 {
-            self.enter_new_round(vec![]);
+        // If we were broadcasting the node for the round already, resume it
+        if let Some(node) = pending_node.filter(|node| node.round() == current_round + 1) {
+            driver.current_round += 1;
+            driver.broadcast_node(node);
+        } else {
+            // kick start a new round
+            let strong_links = driver
+                .dag
+                .read()
+                .get_strong_links_for_round(current_round, &driver.epoch_state.verifier)
+                .unwrap_or(vec![]);
+            driver.enter_new_round(current_round + 1, strong_links);
```

Comment:
Is the assumption that we will have state sync-ed before starting the dag driver? otherwise, this might be a stale round proposal.